src/exchangeRateRouter.ts

```diff
@@ -65,7 +68,7 @@ const v1ExchangeRateIsoAdder: express.RequestHandler = (
   res,
   next
 ): void => {
-  const exReq = asMaybe(asRatesRequest)(req)
+  const exReq = req as ExpressRequest
   if (exReq?.requestedRates == null) return next(500)
```

Comment:
You can consolidate all the null checks into a few common middlewares for each path.